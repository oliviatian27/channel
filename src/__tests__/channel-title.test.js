import React from 'react';
import { shallow } from 'enzyme';

import ChannelTitle from '../components/channel-list/channel-detail/channel-title';


describe('ChannelTitle tests', () => {

  it('renders ChannelTitle component', () => {
    const title = "one";
    const wrapper = shallow(<ChannelTitle title={title} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-title')).toBeDefined();
  });

  it('displays the correct title based on props.title', () => {
    const title = "one";
    const wrapper = shallow(<ChannelTitle title={title} />);

    // Expect the title text to be props.title
    expect(wrapper.find('.channel-title').text()).toBe(title);

    // update the title props
    wrapper.setProps({title:"two"})

    expect(wrapper.find('.channel-title').text()).toBe("two");
  });
})
