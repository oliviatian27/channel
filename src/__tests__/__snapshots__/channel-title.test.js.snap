// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelTitle tests renders ChannelTitle component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelTitle
      title="one"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "one",
        "className": "channel-title",
      },
      "ref": null,
      "rendered": "one",
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "one",
          "className": "channel-title",
        },
        "ref": null,
        "rendered": "one",
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "one",
      "className": "channel-title",
    },
    "ref": null,
    "rendered": "one",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "one",
        "className": "channel-title",
      },
      "ref": null,
      "rendered": "one",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "one",
        "className": "channel-title",
      },
      "ref": null,
      "rendered": "one",
      "type": "div",
    },
  ],
}
`;
